src/Kernel/client/kernel.ts

```diff
@@ -133,10 +133,16 @@ class Kernel {
             "iqsharp_state_dump",
             message => {
                 console.log("my message received", message);
-                //message.content.amplitudes;
-                var htmlGraph = "hi :)";
+                //message.content.state.amplitudes;
+                function printNQubits() {
+                    document.getElementById("ex").innerHTML = message.content.n_qubits;
+                }
+                var htmlGraph = `<button onclick="printNQubits()"> NQubits </button> <p id="ex"> </p>`;
```

Comment:
Using the `onclick` attribute to reference the named function `printNQubits` could cause issues, since it's the same function name for every div that gets printed. My suggestion would be to create buttons as DOM objects so that you can attach unique event handlers to each.

For example:
```JavaScript
let div = document.getElementById(message.content.state.div_id);
if (div !== null) {
    let button = document.createElement("button");
    let p = document.createElement("p");
    button.appendChild(document.createTextNode("NQubits"));
    button.addEventListener("click", event => {
         p.innerHtml = message.content.n_qubits;
    });
    div.appendChild(button);
    div.appendChild(p);
}
```

Reply:
can I still reference the created attributes in strings I set as inner html?

Reply:
Creating elements as DOM objects lets you get to the attributes of elements using methods instead of string processing (https://developer.mozilla.org/en-US/docs/Web/API/Element/setAttribute), which can be helpful to prevent problems caused by improperly escaping strings (escaping _right_ is really hard).